darts/models/torch_forecasting_model.py

```diff
@@ -72,26 +72,31 @@ def _cat_with_optional(tsr1: torch.Tensor, tsr2: Optional[torch.Tensor]):
     def __len__(self):
         return len(self.ts_dataset)
 
-    def __getitem__(self, idx: int) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+    def __getitem__(self, idx: int):
         """
         Cast the content of the dataset to torch tensors
         """
         item = self.ts_dataset[idx]
 
-        if len(item) == 2:
+        if isinstance(self.ts_dataset, TimeSeriesInferenceDataset):
             # the dataset contains (input_target, input_covariate) only
-            input_tgt = torch.from_numpy(item[0]).float()
-            input_cov = torch.from_numpy(item[1]).float() if item[1] is not None else None
-            return self._cat_with_optional(input_tgt, input_cov)
+            input_tgt = torch.from_numpy(item[0].values()).float()
+            input_cov = torch.from_numpy(item[1].values()).float() if item[1] is not None else None
+            future_cov = torch.from_numpy(item[2].values()).float() if item[2] is not None else None
 
-        elif len(item) == 3:
+            if future_cov is not None:
```

Comment:
@hrzn second attempt of solving this issue. The problem with this is we don't have a consistent return type, which is clearly not ideal. At the same time, duplicating the class for every possible scenario seems kind of inefficient too. I don't know yet how to best handle this.

Reply:
I guess this hack can do it for now.